x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/agent/sub_graphs/translate_rule/nodes/translation_result/translation_result.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RuleTranslationResult } from '../../../../../../../../../../common/siem_migrations/constants';
+import type { GraphNode } from '../../types';
+
+export const translationResultNode: GraphNode = async (state) => {
+  // Set defaults
+  const elasticRule = {
+    title: state.original_rule.title,
+    description: state.original_rule.description || state.original_rule.title,
+    severity: 'low',
+    ...state.elastic_rule,
+  };
+
+  const query = elasticRule.query;
+  let translationResult;
+
+  if (!query) {
+    translationResult = RuleTranslationResult.UNTRANSLATABLE;
+  } else {
+    if (query.startsWith('FROM logs-*')) {
+      elasticRule.query = query.replace('FROM logs-*', 'FROM [indexPattern]');
+      translationResult = RuleTranslationResult.PARTIAL;
+    } else if (state.validation_errors?.esql_errors) {
+      translationResult = RuleTranslationResult.PARTIAL;
+    } else if (query.match(/\[(macro|lookup):.*?\]/)) {
+      translationResult = RuleTranslationResult.PARTIAL;
```

Comment:
I think we should not be able to get to this point because it will be caught in the previous check `state.validation_errors?.esql_errors` since `[(macro|lookup):.*?\]` like string will be a syntax error. Does not hurt to have this check though.

Reply:
I changed the validation node to execute the syntax check without the placeholders, this way we only try to the query when there is a real syntax issue. Placeholders were causing the graph to run all iterations trying to fix a query that could not be fixed.
That's why now we check for placeholders explicitly at the end.